packages/ionic/test/number-control.spec.ts

```diff
@@ -0,0 +1,221 @@
+        });
+
+        it('should render integer', () => {
+            renderInteger(
+                fixture,
+                {
+                    data: { foo: 123 },
+                    schema: {
+                        type: 'object',
+                        properties: {
+                            foo: { type: 'integer' }
+                        }
+                    },
+                    uischema: defaultTestData.uischema
+                },
+                () => {
+                    const textInput = fixture.debugElement.query(By.directive(TextInput));
+                    const textInputInstance = textInput.componentInstance;
+                    expect(textInputInstance.value).toBe(123);
+                    expect(textInputInstance.step).toBe(1);
+                    expect(textInputInstance.disabled).toBe(false);
+                });
+        });
+
+        it('should support updating the state', () => {
+            updateFloatState(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                expect(textInputInstance.value).toBe(456.456);
+            });
+        });
+
+        // possibly related to https://github.com/ionic-team/ionic/issues/14370
+        xit('should support updating with undefined', () => {
+            updateWithUndefined(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                // not undefined
+                expect(textInputInstance.value).toBe(undefined);
+            });
+        });
+
+        // possibly related to https://github.com/ionic-team/ionic/issues/14370
+        xit('should support updating with null', () => {
+            updateWithUndefined(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                // not null
+                expect(textInputInstance.value).toBe(null);
+            });
+        });
+
+        it('should not update with wrong ref', () => {
+            updateWithSiblingValue(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                expect(textInputInstance.value).toBe(123.123);
+            });
+        });
+
+        // store needed as we evaluate the calculated enabled value to disable/enable the control
+        // also, see https://github.com/ionic-team/ionic/issues/5280
+        it('can be disabled', () => {
+            canBeDisabled(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
```

Comment:
we could extract:
`const textInput = fixture.debugElement.query(By.directive(TextInput));
const textInputInstance = textInput.componentInstance;`